pkg/process/checks/net.go

```diff
@@ -122,7 +122,11 @@ func (c *ConnectionsCheck) Init(syscfg *SysProbeConfig, hostInfo *HostInfo, _ bo
 	c.processData.Register(c.serviceExtractor)
 
 	// LocalResolver is a singleton LocalResolver
-	c.localresolver = resolver.NewLocalResolver(proccontainers.GetSharedContainerProvider(c.wmeta), clock.New(), maxResolverAddrCacheSize, maxResolverPidCacheSize)
+	sharedContainerProvider, err := proccontainers.GetSharedContainerProvider()
```

Comment:
I don't think this would work for process-agent which this connections check is part of. Looks like `proccontainers.InitSharedContainerProvider` is only called from the agent and cluster-agent binaries. @daniel-taf can you chime in here? I am not sure with the new single agent binary thing if this is a valid concern.

Reply:
IIRC single agent binary is not moving forward (but I could be out of the loop). As it is, this would not work as you say, the process-agent binary should also call `proccontainers.InitSharedContainerProvider`. In fact, I would expect our e2e tests to fail due to this.

Reply:
Nice catch @hmahmood 

I added the call to  `proccontainers.InitSharedContainerProvider` as part of the process-agent boot process.

Reply:
Thanks, could you also remove the field `ConnectionsCheck.wmeta`; it is not used anymore.